addons/l10n_ca/views/res_company_view.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+        <record id="res_company_form_inherit_ca" model="ir.ui.view">
+            <field name="name">res.company.form.inherit.l10n.ca</field>
+            <field name="model">res.company</field>
+            <field name="inherit_id" ref="base.view_company_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='vat']" position="after">
+                    <field name="l10n_ca_pst"/>
```

Comment:
you could put an attrs depending on the account_fiscal_country_id.code (country_code) here as well.  And no problem to keep the others in the layouts.